Kitodo/src/main/java/org/kitodo/production/security/DynamicAuthenticationProvider.java

```diff
@@ -92,6 +92,9 @@ public Authentication authenticate(Authentication authentication) {
                 throw new AuthenticationServiceException("No LDAP server specified on user's LDAP group");
             }
             configureAuthenticationProvider(ldapGroup.getLdapServer().getUrl(), ldapGroup.getUserDN());
+            if (ldapGroup.getUserDN().contains("{ldaplogin}")) {
+                authentication = new UsernamePasswordAuthenticationToken(user.getLdapLogin(), authentication.getCredentials());
+            }
```

Comment:
Why is this change needed? I tested this with a modified user which contains the same values in `login` and `ldaplogin`, the user dn contain the `ldaplogin` placeholder and this user could be authenticated without this change.

Reply:
Yes, of course, if the `ldaplogin` and the `login` contain the same value, it doesn't matter. However, the functionality of the `ldaplogin` is precisely that you can specify a _different_ name here, which is sent to the LDAP _instead of the login_. In the case of a login with a Microsoft Active Directory, this mus usually be “‹Firstname› ‹Lastname›”.

Reply:
Okay. But then I can not verify is this change correct or not nor I can not say if this change is working for the given use case.

Reply:
You can, if you put a fake username (like "foobar") in the `login`, and your login name in the `ldapLogin`. You will see that you can log in with `foobar` and your password.

BTW, it fixed the problem at our customer’s site, who use it productively.